R/zchunk_LB123.Electricity_china.R

```diff
@@ -0,0 +1,262 @@
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L123.in_EJ_province_elec_F",
+             "L123.out_EJ_province_elec_F",
+             "L123.in_EJ_province_ownuse_elec",
+             "L123.out_EJ_province_ownuse_elec"))
+  } else if(command == driver.MAKE) {
+
+    # =============================================================================
+    other <- row <- oil <- natural.gas <- coal<- coal.adj <- wind <- wind.adj<- nuclear<- nuc.adj <-
+      solar <- solar.adj <- biomass <- fuel <- value <- province  <- sector <- year <- copy_rl <- copy_g <-
+      pct <- each <-Sum <- value.x <- value.y <- sector.x <- sector.y <- fuel.x <- fuel.y <- NULL     # silence package check.
+    all_data <- list(...)[[1]]
+    HISTORICAL_YEARS_temp <- c(HISTORICAL_YEARS, 2011, 2012)
+
+    # -----------------------------------------------------------------------------
+    # 1.Load required inputs
+    province_names_mappings     <- get_data(all_data, "gcam-china/province_names_mappings")
+    CPSY_GWh_province_F_elec_out <- get_data(all_data, "gcam-china/CPSY_GWh_province_F_elec_out")
+    L101.inNBS_Mtce_province_S_F <- get_data(all_data, "L101.inNBS_Mtce_province_S_F")
+    L123.in_EJ_R_elec_F_Yh <- get_data(all_data, "L123.in_EJ_R_elec_F_Yh")
+    L123.out_EJ_R_elec_F_Yh <- get_data(all_data, "L123.out_EJ_R_elec_F_Yh")
+    L126.in_EJ_R_elecownuse_F_Yh <- get_data(all_data, "L126.in_EJ_R_elecownuse_F_Yh")
+    L126.out_EJ_R_elecownuse_F_Yh <- get_data(all_data, "L126.out_EJ_R_elecownuse_F_Yh")
+    L132.out_EJ_province_indchp_F <- get_data(all_data, "L132.out_EJ_province_indchp_F")
+
+    # -----------------------------------------------------------------------------
+    # 2.perform computations
+    CPSY_GWh_province_F_elec_out %>%
+      map_province_name(province_names_mappings, "province", TRUE) %>%
+      # In Tibet, electricity in 'other' will be split between oil, gas, solar, and geothermal in 2010. For prior years
+      # it will be moved into coal.
+      mutate(oil = 0,
+             natural.gas = 0,
+             oil = replace(oil, province == "XZ" & year == 2010, other[province == "XZ" & year == 2010] / 4),
+             natural.gas = replace(natural.gas, province == "XZ" & year == 2010, other[province == "XZ" & year == 2010] / 4),
+             coal = coal + coal.adj,
+             wind = wind + wind.adj,
+             nuclear = nuclear + nuc.adj,
+             solar = solar + solar.adj) %>%
+      select(-other, -row, -coal.adj, -wind.adj, -nuc.adj, -solar.adj) %>%
+      mutate(biomass = coal) %>%
+      gather(fuel, value, -province, -year) %>%
+      filter(value == 0 | value > 0.1) %>%
+      complete(nesting(province, fuel), year = HISTORICAL_YEARS_temp) %>%
+      arrange(province, fuel, year) %>%
+      group_by(province, fuel) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup %>%
+      mutate(fuel = as.character(fuel)) ->
+      L132.GWh_province_F_elec_out
```

Comment:
L123?

Reply:
It's L132.GWh_province_F_elec_out in the old data system, so just keep same in the new.